backend/src/main/java/com/company/project/service/TimetableService.java

```diff
@@ -82,4 +94,22 @@ public List<TimetableDto> updateTimeslots(List<Timeslot> timeslotDtos) {
         return timeSlotListToTimetableList(updatedTimeslots);
     }
 
+
+    public ShareLinkDto createShareLink(HttpServletRequest request) throws URISyntaxException {
+        String host = request.getRequestURL().toString();
+        String link = host.substring(0, host.lastIndexOf(new URI(host).getPath()));
+
+        ShareLink savedLink = activeLinkRepository.save(new ShareLink(link + "/students/timetable"));
+        return new ShareLinkDto(savedLink.getShareLink());
+    }
```

Comment:
Zastanawiam się czy jest to w ogóle potrzebne. Link równie dobrze mógłby być jakąś podstroną na froncie, potrzebowalibyśmy tylko endpoint na serwerze stwierdzający czy link jest aktywny. Można zamiast linka generować też numer sesji, co może nam się przyda jeśli się wyrobimy

Reply:
Ja bym powiedział, że jeśli chcemy sobie zamknąć całkowicie możliwość dodania wielu sesji, to jak najbardziej można hardcodować link na froncie, w bazie zrobić tabelę Link(id, isActive) z jednym rekordem (xd). Jeżeli ten kod komukolwiek dodaje pracy z sharowaniem i koszt zmiany tego kodu jest mniejszy niż zysk z tejże zmiany, to uważam, że można to wyrzucić.

Reply:
Proponuję w takim razie zostawić tak jak jest i zmergować. Jak nie będzie się podobało ludziom z frontu to wtedy zmienimy

Reply:
Jasne :)